symfony/plugins/orangehrmAdminPlugin/test/Dao/UserDaoTest.php

```diff
@@ -0,0 +1,221 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Admin\Tests\Dao;
+
+use OrangeHRM\Admin\Dao\UserDao;
+use OrangeHRM\Authentication\Dto\UserCredential;
+use OrangeHRM\Config\Config;
+use OrangeHRM\Core\Exception\DaoException;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\Entity\User;
+use OrangeHRM\Entity\UserRole;
+use OrangeHRM\ORM\Doctrine;
+use OrangeHRM\Tests\Util\TestCase;
+use OrangeHRM\Tests\Util\TestDataService;
+
+/**
+ * @group Admin
+ * @group Dao
+ */
+class UserDaoTest extends TestCase
+{
+
+    private $systemUserDao;
+    protected $fixture;
+
+    /**
+     * Set up method
+     */
+    protected function setUp(): void
+    {
+        $this->systemUserDao = new UserDao();
```

Comment:
We should use snapshots for testing .. rather than delete/insert stuff

Reply:
for the fixtures loading?